patches/graphql-relay+0.5.3.patch

```diff
@@ -0,0 +1,31 @@
+patch-package
+--- a/node_modules/graphql-relay/lib/connection/arrayconnection.js
++++ b/node_modules/graphql-relay/lib/connection/arrayconnection.js
+@@ -61,7 +61,8 @@ function connectionFromArraySlice(arraySlice, args, meta) {
+       first = args.first,
+       last = args.last;
+   var sliceStart = meta.sliceStart,
+-      arrayLength = meta.arrayLength;
++      arrayLength = meta.arrayLength,
++      resolveNode= meta.resolveNode;
+ 
+   var sliceEnd = sliceStart + arraySlice.length;
+   var beforeOffset = getOffsetWithDefault(before, arrayLength);
+@@ -88,10 +89,15 @@ function connectionFromArraySlice(arraySlice, args, meta) {
+   var slice = arraySlice.slice(Math.max(startOffset - sliceStart, 0), arraySlice.length - (sliceEnd - endOffset));
+ 
+   var edges = slice.map(function (value, index) {
+-    return {
++    const newEdge = {
+       cursor: offsetToCursor(startOffset + index),
+-      node: value
++      node: !!resolveNode ? Promise.resolve(resolveNode(value)) : value,
```

Comment:
Oh, so I can actually safely remove the `Promise.resolve` wrapping. I injected a `debugger` with a custom `then: () => {}` object and saw the call stack:

<img width="414" alt="screen shot 2018-01-23 at 1 05 00 pm" src="https://user-images.githubusercontent.com/1457859/35292284-06347690-003e-11e8-8de0-6859787b0e20.png">

`completeValue` is recursive so I just pasted a bit.

Essentially, GraphQL handles 'complete values' or promises interchangably for fields, so in this case, the Relay library is just munging data around and passing it through, so it's safe here to just pass through the resolve callback, we can drop `Promise.resolve`, which is redundant.

Reply:
Ace 👌 Knowing is half the battle!